### PR TITLE
cwalk: new package

### PR DESCRIPTION
**Pull request by antoniovazquezblanco**

### CODE REVIEW

File: subprojects/packagefiles/cwalk/meson.build [hunk 1]
```diff
@@ -0,0 +1,15 @@
+project('cwalk', 'c',
+  license: 'MIT',
+  meson_version: '>= 0.57.0'
```
Comment (neheb) [inline subprojects/packagefiles/cwalk/meson.build:3]: pretty sure you can reduce to 0.49.0

Comment (antoniovazquezblanco) [reply subprojects/packagefiles/cwalk/meson.build:3]: ..\subprojects\cwalk-1.2.7\meson.build:15: WARNING: Project targets '>= 0.49.0' but uses feature introduced in '0.50.0': include_directories kwarg of type string. Use include_directories('include') instead

**Comment by antoniovazquezblanco** [comment]
A release was pushed with fixed meson version.

https://github.com/likle/cwalk/issues/42

**Comment by antoniovazquezblanco** [comment]
It seems the default option for `default_libraries` is `shared`. See https://mesonbuild.com/Builtin-options.html

For some reason, Visual Studio check script is checking for the presence of the static one... Something wrong on my end?

Thanks!

**Comment by neheb** [comment]
```
ERROR: File 'subprojects/cwalk-1.2.8/cwalk.lib' could not be found
```

is caused by no functions being exported. This can be fixed with a custom def file.

**Comment by antoniovazquezblanco** [comment]
This is being fixed upstream https://github.com/likle/cwalk/pull/46

**Comment by antoniovazquezblanco** [comment]
Now all tests pass! :D

File: subprojects/packagefiles/cwalk/meson.build [hunk 2]
```diff
@@ -0,0 +1,29 @@
+project('cwalk', 'c',
```
Comment (jpakkane) [inline subprojects/packagefiles/cwalk/meson.build:1]: Upstream already has a `meson.build` file. Could this not use that one instead?

Comment (antoniovazquezblanco) [reply subprojects/packagefiles/cwalk/meson.build:1]: It needs a fix that has just merged. We will be able to use upstream meson file in the next version.

File: subprojects/packagefiles/cwalk/meson.build [hunk 3]
```diff
@@ -0,0 +1,29 @@
+project('cwalk', 'c',
+  license: 'MIT',
+  version: '1.2.9',
+  meson_version: '>= 0.57.0'
+)
+
+cwalk_inc = include_directories('include')
+
+cwalk_c_args = []
+if get_option('default_library') == 'shared'
```
[the remark this answers is not in the document]
Comment (antoniovazquezblanco) [reply subprojects/packagefiles/cwalk/meson.build]: Done! Thanks

**Comment by antoniovazquezblanco** [comment]
Upstream fix at https://github.com/likle/cwalk/pull/47

**Comment by antoniovazquezblanco** [comment]
From upstream, @likle asks:

> I am not verify familiar with meson, doesn't "both" mean it builds shared as well as static?
> Wouldn't that set it to CWK_SHARED for the static build done using the both option?

**Comment by antoniovazquezblanco** [comment]
Ping @neheb :)

**Comment by antoniovazquezblanco** [comment]
Thanks @neheb!

Do you have an answer for https://github.com/mesonbuild/wrapdb/pull/1387#issuecomment-2098203749 ?

How does meson deal with dual builds?

Thanks again!!

**Comment by neheb** [comment]
both builds both shared and static. When using in a dependency, the shared is used. In wraps around here, typically != static is checked to enable a SHARED macro.

**Comment by benoit-pierre** [comment]
IMHO, wraps that need special shared/static arguments should use `<lang>_shared_args` / `<lang>_static_args` when available and otherwise error on meson < 1.3.0 when building with `--default-library=both`.

**Comment by neheb** [comment]
Right. It’s usually not an issue with GCC based compilers but not MSVC. The best solution for MSVC currently is to avoid these and use def files.

**Pull request merged**